### PR TITLE
Adding serializer/deserializer for sqlblob

### PR DESCRIPTION
**Pull request by rsatrio**
Hi,
My previous PR is missing because I delete my commit. Anyway, this PR is for serializer/deserializer SqlBlob into Base64 String. This will be very useful in cases send query result through json.  What I have done since the last commit are:
1. Move the serializer/deserializer registration into OptionalHandlerFactory (following your example in your latest commit),
2. Refactor to use Jackson's Base64 Function,
3. Move the unit test to new class
4. I still put SerialBlob because it is the standard implementation of Sql Blob
5. Sent CLA doc via email

Thanks,

Rizky

### CODE REVIEW

File: src/main/java/com/fasterxml/jackson/databind/deser/std/SqlBlobDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package com.fasterxml.jackson.databind.deser.std;
+
+import java.io.IOException;
+import java.sql.Blob;
+import java.util.Arrays;
+import java.util.UUID;
+
+import javax.sql.rowset.serial.SerialBlob;
```
Comment (cowtowncoder) [inline src/main/java/com/fasterxml/jackson/databind/deser/std/SqlBlobDeserializer.java:8]: Hmmmh. I wonder... this seems to require dependency to another JDK9+ module

https://docs.oracle.com/en/java/javase/11/docs/api/java.sql.rowset/javax/sql/rowset/serial/package-summary.html

beyond basic `java.sql`. Will need to add to Moditect sources I think

**Comment by cowtowncoder** [comment]
Looks good, will merge. Thank you!

**Comment by cowtowncoder** [comment]
Ended up dropping deserialization support from 2.12, as I think it gets tricky, requires new Java 9 module dependency, and has not yet been requested. There are also some technical things about deserialization itself to consider (it's not just json, so incoming value may be "true" binary or base64-encoded data etc). So will want to wait for it being requested -- should probably not have suggested addition yet.
Serialization part may need improvement too, as actual `Blob` implementations may not get mapped... but user feedback can help there.

**Comment by rsatrio** [comment]
Dear Tatu,
Noted, I agree with you, the deserialization method I've made is too
specific for Base64 to Blob. Just a little suggestion from me, maybe you
can still keep the SqlBlobDeserialization class? So users can still use it
via @JsonDeserialize.
Any other feature that I can help? Or maybe I can help to add some unit
test?

Thanks,

Rizky

**Comment by cowtowncoder** [comment]
I think at this point users will need to have a look at PR, if they need to add it. If there was a convenient way to add example code I would include it; but otherwise I don't think I want to add unused code. That tends to lead to maintenance challenges.
But like I said, if someone actually requests deserializer (support for this type has never been requested before that), a new implementation may be added.

**Pull request merged**